static/js/tools/timeline/data_fetcher.ts

```diff
@@ -292,3 +293,74 @@ export function fetchStatData(
     return result;
   });
 }
+
+/**
+ * Creates a new StatData object for all measurement methods of the stat var,
+ * with an artificial stat var in the form of StatVar-MMethod. The StatData
+ * values for the StatVar is also updated to be the mean across the
+ * measurement methods for the data.
+ *
+ * @param mainStatData StatData for the main data line to plot.
+ * @param modelStatAllResponse StatAllApiResponse for all available measurement methods for the mainStatData.
+ *
+ * @return pair of (transformed mainStatData with mean values, processed model StatData)
+ */
+export function statDataFromModels(
+  mainStatData: StatData,
+  modelStatAllResponse: StatAllApiResponse
+): [StatData, StatData] {
+  const modelData = {
+    places: [],
+    statVars: [],
+    dates: [],
+    data: {},
+    sources: new Set<string>(),
+    measurementMethods: new Set<string>(),
+  };
+  if (!mainStatData.dates.length) {
+    return [mainStatData, modelData];
+  }
+  for (const place in modelStatAllResponse.placeData) {
+    const placeData = modelStatAllResponse.placeData[place];
+    modelData.places.push(place);
+    modelData.data[place] = { data: {} };
+    for (const sv in modelStatAllResponse.placeData[place].statVarData) {
+      const mainObsPeriod =
+        mainStatData.data[place].data[sv].metadata.observationPeriod;
+      modelData.statVars.push(sv);
+      const svData = placeData.statVarData[sv];
+      if ("sourceSeries" in svData) {
+        const means = {};
+        const sourceSeries = svData.sourceSeries;
+        // Replace requested series with means across models.
+        const keepSeries = [];
+        for (const series of sourceSeries) {
+          if (series.observationPeriod !== mainObsPeriod) {
+            continue;
+          }
+          keepSeries.push(series);
+          mainStatData.measurementMethods.delete(series.measurementMethod);
+          modelData.measurementMethods.add(series.measurementMethod);
+          modelData.sources.add(series.provenanceDomain);
+          for (const date of Object.keys(series.val).sort()) {
+            means[date] = date in means ? means[date] : [];
+            means[date].push(series.val[date]);
+          }
+          const newSv = `${sv}-${series.measurementMethod}`;
+          modelData.data[place].data[newSv] = { val: series.val };
+          modelData.statVars.push(newSv);
+        }
+        for (const date in means) {
+          means[date] = _.mean(means[date]);
+        }
+        svData.sourceSeries = keepSeries;
```

Comment:
looks like this would be updating modelStatAllResponse, is that what you wanted? If so, maybe add a comment to explain this part

Reply:
added a comment